aspnet/fundamentals/routing.rst

```diff
@@ -1,10 +1,285 @@
+	  - 4.234
+	  - Matches a valid ``double`` value
+	* - ``float``
+	  - {price:float}
+	  - 3.14
+	  - Matches a valid ``float`` value
+	* - ``guid``
+	  - {id:guid}
+	  - 7342570B-44E7-471C-A267-947DD2A35BF9
+	  - Matches a valid ``Guid`` value
+	* - ``long``
+	  - {ticks:long}
+	  - 123456789
+	  - Matches a valid ``long`` value
+	* - ``minlength(value)``
+	  - {username:minlength(5)}
+	  - steve
+	  - String must be at least 5 characters long.
+	* - ``maxlength(value)``
+	  - {filename:maxlength(8)}
+	  - somefile
+	  - String must be no more than 8 characters long.
+	* - ``length(min,max)``
+	  - {filename:length(4,16)}
+	  - Somefile.txt
+	  - String must be at least 8 and no more than 16 characters long.
+	* - ``min(value)``
+	  - {age:min(18)}
+	  - 19
+	  - Value must be at least 18.
+	* - ``max(value)``
+	  - {age:max(120)}
+	  - 91
+	  - Value must be no more than 120.
+	* - ``range(min,max)``
+	  - {age:max(18,120)}
+	  - 91
+	  - Value must be at least 18 but no more than 120.
+	* - ``alpha``
+	  - {name:alpha}
+	  - Steve
+	  - String must consist of alphabetical characters.
+	* - ``regex(expression)``
+	  - {ssn:regex(\d{3}-\d{2}-\d{4})}
+	  - 123-45-6789
+	  - String must match the provided regular expression.
+	* - ``required``
+	  - {name:required}
+	  - Steve
+	  - Used to enforce that a non-parameter value is present during during URL generation.
+
+Inline constraints must match one of the above options, are an exception will be thrown.
+
+.. tip:: To constrain a parameter to a known set of possible values, you can use a regex: ``{action:regex(list|get|create)}``. This would only match the ``action`` route value to ``list``, ``get``, or ``create``. If passed into the constraints dictionary, the string "list|get|create" would be equivalent. Constraints that are passed in the constraints dictionary (not inline within a template) that don't match one of the known constraints are also treated as regular expressions.
+
+.. warning:: Using constraints for **validation** means that invalid input will result in a 404 (Not Found) instead of a 400 with an appropriate error message. Route constraints should be used to **disambiguate** between routes, not validate the inputs for a particular route.
+
+Constraints can be *chained*. You can specify that a route value is of a certain type and also must fall within a specified range, for example: ``{age:int:range(1,120)}``.  Numeric constraints like ``min``, ``max``, and ``range`` will automatically convert the value to ``long`` before being applied unless another numeric type is specified.
+
+Route templates must be unambiguous, or they will be ignored. For example, ``{id?}/{foo}`` is ambiguous, because it's not clear which route value would be bound to a request for "/bar". Similarly, ``{*everything}/{plusone}`` would be ambiguous, because the first route parameter would match everything from that part of the request on, so it's not clear what the ``plusone`` parameter would match.
```

Comment:
Yeah, this is a better description, but it's not ignored. Route templates like those will result in an exception when you call `MapRoute`